frontend/public/components/modals/delete-pvc-modal.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { Alert, Stack, StackItem } from '@patternfly/react-core';
+import { AsyncComponent, HandlePromiseProps, withHandlePromise } from '../utils';
+import { getName, YellowExclamationTriangleIcon } from '@console/shared';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '../factory';
+import { k8sKill, PersistentVolumeClaimKind } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '../../models';
+import { isPVCDelete, PVCDelete, useExtensions } from '@console/plugin-sdk';
+
+const DeletePVCModal = withHandlePromise<DeletePVCModalProps>((props) => {
+  const { pvc, inProgress, errorMessage, handlePromise, close, cancel } = props;
+  const pvcDeleteExtensions = useExtensions<PVCDelete>(isPVCDelete);
+  const pvcName = getName(pvc);
+
+  const submit = (e) => {
+    e.preventDefault();
+
+    const promise = k8sKill(PersistentVolumeClaimModel, pvc);
+    const extensionPromises = pvcDeleteExtensions.map(
+      ({ properties: { predicate, onPVCKill } }) => predicate(pvc) && onPVCKill(pvc),
+    );
+    return handlePromise(Promise.all([promise, ...extensionPromises]), close);
+  };
+
+  const alertComponents = pvcDeleteExtensions.map(
+    ({ properties: { predicate, alert }, uid }) =>
+      predicate(pvc) && (
+        <Alert
```

Comment:
If you wrap the alert in `StackItem` you wont need to `map` again later.

```suggestion
       <StackItem key={uid}>
          <Alert
```
you can also move key from `Alert` to `StackItem`. There's also no need to make `key` more complicated than just uid. Doing `pvc-alert-${uid}` doesnt bring any value